app/controllers/internal/events_controller.rb

```diff
@@ -14,8 +18,75 @@ def show
       @page_title = @event.name
     end
 
+    def new
+      @event = Event.new(venue_type: Event::VENUE_TYPES[:existing])
+      @event.building = EventBuilding.new
+    end
+
+    def approve
+      api_event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:format])
```

Comment:
Why is this `params[:format]`? I would expect it to be something like `params[:id]`